File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<string> s_num;
bool cmp(string a, string b);
string solution(vector<int> numbers) {
    string answer = "";
    for(int i : numbers) s_num.push_back(to_string(i));
    sort(s_num.begin(), s_num.end(), cmp);
    for(string s : s_num) answer+=s;
    
    bool chk = true;
    for(char c : answer){
        if(c != '0') {
            chk = false;
            break;
        }
    }
    
    if(chk) answer="0";
    
    return answer;
}
bool cmp(string a, string b){
    return a+b > b+a;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] sArr = new String[numbers.length];
        
        for(int i=0; i<numbers.length; i++) sArr[i] = Integer.toString(numbers[i]);
        Arrays.sort(sArr, (a,b) -> (b+a).compareTo(a+b));
        for(String t : sArr) answer+=t;
        
        if(sArr[0].equals("0")) return "0";
        return answer;
    }
}